sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/ToString.java

```diff
@@ -18,51 +18,176 @@
- *
- * <p>Example of use:
- * <pre> {@code
- * PCollection<Long> longs = ...;
- * PCollection<String> strings = longs.apply(ToString.<Long>element());
- * } </pre>
- *
+ * {@link PTransform PTransforms} for converting a {@link PCollection PCollection&lt;?&gt;},
+ * {@link PCollection PCollection&lt;KV&lt;?,?&gt;&gt;}, or
+ * {@link PCollection PCollection&lt;Iterable&lt;?&gt;&gt;}
+ * to a {@link PCollection PCollection&lt;String&gt;}.
  *
  * <p><b>Note</b>: For any custom string conversion and formatting, we recommend applying your own
  * {@link SerializableFunction} using {@link MapElements#via(SerializableFunction)}
  */
 public final class ToString {
+  private ToString() {
+    // do not instantiate
+  }
 
   /**
    * Returns a {@code PTransform<PCollection, PCollection<String>>} which transforms each
    * element of the input {@link PCollection} to a {@link String} using the
    * {@link Object#toString} method.
    */
-  public static PTransform<PCollection<?>, PCollection<String>> element() {
-    return new Default();
+  public static PTransform<PCollection<?>, PCollection<String>> of() {
+    return new SimpleToString();
   }
 
-  private ToString() {
+  /**
+   * Returns a {@code PTransform<PCollection, PCollection<String>>} which transforms each
+   * element of the input {@link PCollection} to a {@link String} using the
+   * {@link Object#toString} method.
+   */
+  public static PTransform<PCollection<? extends KV<?, ?>>, PCollection<String>> kv() {
+    return kv(",");
+  }
+
+  /**
+   * Returns a {@code PTransform<PCollection, PCollection<String>>} which transforms each
+   * element of the input {@link PCollection} to a {@link String} using the
+   * {@link Object#toString} method.
+   * @param delimiter The delimiter to put between the key and value
+   */
+  public static PTransform<PCollection<? extends KV<?, ?>>,
+          PCollection<String>> kv(String delimiter) {
+    return new KVToString(delimiter);
+  }
+
+  /**
+   * Returns a {@code PTransform<PCollection, PCollection<String>>} which transforms each
+   * element of the input {@link PCollection} to a {@link String} using the
+   * {@link Object#toString} method.
+   */
+  public static PTransform<PCollection<? extends Iterable<?>>, PCollection<String>> iterable() {
+    return iterable(",");
+  }
+
+  /**
+   * Returns a {@code PTransform<PCollection, PCollection<String>>} which transforms each
+   * element of the input {@link PCollection} to a {@link String} using the
+   * {@link Object#toString} method.
+   * @param delimiter The delimiter to put between the key and value
+   */
+  public static PTransform<PCollection<? extends Iterable<?>>,
+          PCollection<String>> iterable(String delimiter) {
+    return new IterablesToString(delimiter);
   }
 
   /**
    * A {@link PTransform} that converts a {@code PCollection} to a {@code PCollection<String>}
    * using the {@link  Object#toString} method.
+   *
+   * <p>Example of use:
+   * <pre> {@code
```

Comment:
no space between `<pre> {@code` and same on the other end. These show as whitespace in the rendered javadoc.